### PR TITLE
Use in xDAI blockscout blockexplorer link and refactor

### PR DESCRIPTION
**Pull request by anxolin**

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
   * **🎩 [Dapp](https://pr1430--dexreact.review.gnosisdev.com)**: Testing web app
   * **📚 [Storybook](https://pr1430--dexreact.review.gnosisdev.com/storybook/)**: Component stories

File: src/components/common/BlockExplorerLink.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+   */
+  className?: string // to allow subclassing styles with styled-components
+}
+
+function getEtherscanUrlPrefix(networkId: Network): string {
+  return !networkId || networkId === Network.Mainnet ? '' : (Network[networkId] || '').toLowerCase() + '.'
+}
+
+function getEtherscanUrlSuffix(type: BlockExplorerLinkType, identifier: string): string {
+  switch (type) {
```
Comment (Velenir) [inline src/components/common/BlockExplorerLink.tsx:41]: Does TS not complain about no `default`?

File: src/components/common/BlockExplorerLink.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  const { networkIdOrDefault: networkIdWallet } = useWalletConnection()
+
+  const networkId = networkIdFixed || networkIdWallet
+
+  if (!networkId || !identifier) {
+    return null
+  }
+
+  const url = getExplorerUrl(networkId, type, identifier)
+  const label = labelProp || (useUrlAsLabel ? url : undefined) || abbreviateString(identifier, 6, 4)
```
Comment (Velenir) [inline src/components/common/BlockExplorerLink.tsx:111]:
```suggestion
  const label = labelProp || (useUrlAsLabel && url) || abbreviateString(identifier, 6, 4)
```

File: src/components/common/BlockExplorerLink.tsx (hunk 1 above)
Comment (anxolin) [reply src/components/common/BlockExplorerLink.tsx:41]: No, that's something I liked in this case about TS

It knows all possible types, and it knows I'm being exhaustive, so it knows I always return a string
If I delete any of the cases, then he complains

Comment (Velenir) [reply src/components/common/BlockExplorerLink.tsx:41]: TS smart :mortar_board:

**Review by Velenir — APPROVED** [review]

**Comment by Velenir** [comment]
Travis: `Property 'xDai' does not exist on type 'typeof Network'. Did you mean 'xDAI'?`

**Pull request merged**